remote-droid-guard/src/main/java/org/microg/gms/droidguard/DroidguardHelper.java

```diff
@@ -89,6 +90,31 @@
         return invoke(context, clazz, request.packageName, request.reason, response.byteCode.toByteArray(), request.androidIdLong, request.extras);
     }
 
+    public static int gmsVersionCode(String arch) {
+        switch (arch) {
+            case "arm64-v8a":
+                return 12688023;
+            case "x86":
+                return 12688026;
+            case "x86_64":
+                return 12688027;
+            default:
+                return 12688019;
+        }
+    }
+
+    public static String gmsVersionNamePrefix(String arch) {
+        switch (arch) {
+            case "arm64-v8a":
+                return "12.6.88 (040400-{{cl}})";
```

Comment:
What is that `{{cl}}` thing?

Reply:
Can it be a placeholder that is later replaced in the code?
Have you tried to sniff the real web request?

Reply:
It could, but it works as is. I haven't checked a real request.